### PR TITLE
Add Docker step to Pre-Release Checklist

### PR DESCRIPTION
**Pull request by lidel**
Suggested in https://github.com/ipfs/go-ipfs/issues/3999#issuecomment-392748363 to avoid problem described in https://github.com/ipfs/go-ipfs/issues/3999#issuecomment-392747159

cc @whyrusleeping @VictorBjelkholm

### CODE REVIEW

File: docs/releases.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ will be allowed up to release day.
 - [ ] publish dist.ipfs.io
 - [ ] publish next version to https://github.com/ipfs/npm-go-ipfs
 - [ ] publish gx version with `gx release`, as per [gx release guidelines](https://github.com/whyrusleeping/gx#publishing-and-releasing)
+- [ ] publish image to [Docker Hub](https://hub.docker.com/r/ipfs/go-ipfs/tags/) and update `latest` and `release` tags to point at it
```
Comment (VictorBjelkholm) [inline docs/releases.md]: The publish should be automatic, but I think it takes a while. We should build image on one machine but then just update the tags. 

I think we can setup the automatic build in Docker Hub to be automatic and tag images based on branches... That way, we can have a `release` branch that would point to the `release` tag in Docker Hub, and you'll only need to update that and push it to update the Docker Hub tags.

Comment (lidel) [reply docs/releases.md]: While we are at this: are we able to update or remove the `latest` tag?
IMO it is too ambiguous, and keeping `release` and `master` may be a better idea.
Alternatively it should be automated to point at the same things as `release`

Comment (VictorBjelkholm) [reply docs/releases.md]: `latest` is de-facto (or maybe just is) standard in Docker-land, so we should probably keep it. Having `latest` being in sync with `release` I think is what we've agreed before and would map best to how things usually are dealt with. `master` would be the "nightly" basically with all the latest changes.

**Comment by whyrusleeping** [comment]
what is actually involved in publishing an image to dockerhub? updating the release branch on the repo is already part of the process (i just forget sometimes :/ )

**Comment by olizilla** [comment]
As @whyrusleeping said, Dockerhub has been configured to build and tag the docker images when the master and release branches change on the go-ipfs repo.

The intergration is currerntly broken, pending reconnecting github and dockerhub, but that'll get fixed.
https://github.com/ipfs/go-ipfs/pull/5043#issuecomment-493003631

The releaser doesn't have to perform any extra steps, so I'm gonna close this out.

**Pull request closed without merging**